### PR TITLE
[AC-438] Upgrade to support latest Android Version

### PR DESCRIPTION
**Pull request by f4ww4z**
https://issues.openmrs.org/browse/AC-438

Upgrade compileSdkVersion and targetSdkVersion to the latest API level (27).

### CODE REVIEW

**Comment by Sub6Resources** [comment]
@f4ww4z Update the yml's for CI to version 27 as well and then the checks will pass.

**Comment by f4ww4z** [comment]
I'm using the new `sdkmanager` to download the Android SDK 27, build tools `27.0.2`, and other tools used before, because `android` command errors.

**Comment by f4ww4z** [comment]
I will need a few tries before getting it to work.

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
-  components:
-  - tools
-  - platform-tools
-  - build-tools-23.0.2
-  - android-23
-  - extra-android-m2repository
+
+before_install:
+  - echo "y" | sdkmanager "platforms;android-27" "build-tools;23.0.2" "tools" "platform-tools" "extras;google;m2repository"
+
```
Comment (Sub6Resources) [inline .travis.yml]: The android command should work just fine. Is there a certain part that keeps failing that prompted you to switch?

**Comment by Sub6Resources** [comment]
@f4ww4z the changes I made on AC-437 should fix your lint errors. After my PR is merged, make sure to update yours because it should fix the lint error that is causing your build to fail.

File: .travis.yml (hunk 1 above)
Comment (f4ww4z) [reply .travis.yml]: Just saw the android command works on your PR. Will update when i get home. Waiting for your PR to get merged first though.

**Comment by f4ww4z** [comment]
@Sub6Resources Yup, waiting for your PR to be merged. Will you update the appcompat versions in your PR?

**Comment by Sub6Resources** [comment]
@f4ww4z, no, because lint requires that the Android version match the
support library version. Include it in your PR or create a new issue after
yours is merged.

On Jan 12, 2018 5:12 PM, "Maharaj Fawwaz A. Yusran" <
notifications@github.com> wrote:

@Sub6Resources <https://github.com/sub6resources> Yup, waiting for your PR
to be merged. Will you update the appcompat versions in your PR?

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/openmrs/openmrs-contrib-android-client/pull/404#issuecomment-357389477>,
or mute the thread

**Comment by dkayiwa** [comment]
@f4ww4z did you address the above comments?

**Comment by f4ww4z** [comment]
I am trying again without `sdkmanager` and realized android-27 license agreement is somehow different. See https://stackoverflow.com/a/47726910/8434188

**Comment by Sub6Resources** [comment]
@f4ww4z you're right. Use the workaround given there and hopefully travis will fix the issue.

**Comment by f4ww4z** [comment]
So the travis build is **working** (it currently errors because of `lint` issues which are fixed in #405 ). Now I need to call `sdkmanager` in appveyor, but I am unable to call it as it returns

```
"C:\android-sdk-windows\tools\bin\sdkmanager.bat" is not recognized as an internal or external command,
operable program or batch file.
```
I tried `bin\sdkmanager.sh` and `bin\sdkmanager`, and they all won't work.
@Sub6Resources what is the way to call `sdkmanager` in appveyor? @dkayiwa Are you able to help? Thanks. I think after we can build in appveyor this PR can be merged.

**Comment by f4ww4z** [comment]
I think the appveyor build is working now. It is downloading a newer version of Android SDK `tools` and and I have updated the command accordingly (using the new `sdkmanager.bat`).

File: openmrs-client/build.gradle [hunk 2]
```diff
@@ -38,7 +38,7 @@ android {
         versionCode buildNumber
 
         minSdkVersion 16
-        targetSdkVersion 23
```
Comment (Sub6Resources) [inline openmrs-client/build.gradle]: Can you update the support libraries to version 27 as well? I believe that falls into the scope of this issue.

Comment (f4ww4z) [reply openmrs-client/build.gradle]: Alright.

**Comment by DefCon-007** [comment]
@f4ww4z  Did you get time to look into this ? :)

**Comment by f4ww4z** [comment]
Hi @DefCon-007, I completed the requirements of this PR, you can see that the builds are failing because of lint errors, and they are fixed at #405 .

**Comment by f4ww4z** [comment]
@dkayiwa, is this ready to be merged?

**Comment by dkayiwa** [comment]
I will merge as soon as the main reviewers approve. 😊

**Pull request merged**